initfiles/sbin/hpcc_setenv.in

```diff
@@ -23,30 +23,39 @@
 ###<REPLACE>###
 
 function kill_process () {
-    SENTINEL=$1
-    PID=$2
-    TIMEOUT=$3
-    if [ -e $SENTINEL ]; then
-        rm -f $SENTINEL
+    PID=$1
+    TIMEOUT=$2
+    if [[ $# -eq 3 ]]; then
+      SENTINEL=$3
+      [[ -e $SENTINEL ]] && rm -f $SENTINEL
     fi
-    if [ -e $PID ]; then
+    if [[ -e $PID ]]; then
         pidwait_fn $PID $TIMEOUT
     fi
 }
 
 function pidwait_fn () {
     PID=$1
-    WATCH_PID=`cat $PID`
+    WATCH_PID=$( cat $PID )
     TIMEOUT=$(($2*1000))
+    COMM=$( ps -q $WATCH_PID -o command= )
     kill $WATCH_PID
-    while [ -d /proc/$WATCH_PID -a $TIMEOUT -gt 0 ]; do
-            sleep 0.2
-            TIMEOUT=$(($TIMEOUT-200))
+    while [[ -d /proc/$WATCH_PID ]] && [[ $TIMEOUT -gt 0 ]]; do
+      sleep 0.2
+      TIMEOUT=$(($TIMEOUT-200))
     done
-    if [ $TIMEOUT -le 0 ]; then
-            kill -9 $WATCH_PID
+    COMM_CHECK=$( ps -q $WATCH_PID -o command= )
+    if [[ $TIMEOUT -le 0 ]] && [[ $COMM == $COMM_CHECK ]]; then
```

Comment:
yes it should be -p. All that matters is that the entire string from COMM is the same as what we get from COMM_CHECK, so multiple columns is fine.